### ISSUE TITLE
[FEATURE] Integrate BingWebSeach

### ISSUE THREAD

**Issue by omarmahamid** [langchain4j/langchain4j#2167]
Integrate 

https://learn.microsoft.com/en-us/bing/search-apis/bing-web-search/quickstarts/sdk/web-search-client-library-java

**Comment by omarmahamid** [comment]
@langchain4j you can assign it to me

**Comment by langchain4j** [comment]
@omarmahamid please note that Bing web search is not very high in the priorities, perharps we should finish https://github.com/langchain4j/langchain4j/pull/2124 and other PRs first?

**Comment by omarmahamid** [comment]
Yes, I will finish the remain things (watsonx, llama3) and start with this
